File: app/src/test/java/com/smlnskgmail/jaman/remotetemperaturecontrol/datasend/data/TestSignalTarget.java
package com.smlnskgmail.jaman.remotetemperaturecontrol.datasend.data;

import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.MonitorTarget;
import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.entities.signaltype.SignalTarget;
import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.entities.signaltype.SignalType;
import org.jetbrains.annotations.NotNull;

public class TestSignalTarget implements SignalTarget {

    private MonitorTarget monitorTarget;

    public TestSignalTarget(MonitorTarget monitorTarget) {
        this.monitorTarget = monitorTarget;
    }

    @Override
    public void onNewDataAvailable(@NotNull SignalType signalType, @NotNull String data) {
        switch (signalType) {
            case Temperature:
                monitorTarget.temperatureAvailable(data);
                break;
            case TemperatureMinimum:
                monitorTarget.temperatureMinimumAvailable(data);
                break;
            case TemperatureMaximum:
                monitorTarget.temperatureMaximumAvailable(data);
                break;
            case Humidity:
                monitorTarget.humidityAvailable(data);
                break;
            case HumidityMinimum:
                monitorTarget.humidityMinimumAvailable(data);
                break;
            case HumidityMaximum:
                monitorTarget.humidityMaximumAvailable(data);
                break;
            case Reset:
                monitorTarget.resetRequired();
                break;
        }
    }

}

File: app/src/test/java/com/smlnskgmail/jaman/remotetemperaturecontrol/datasend/data/TestSignalTarget.kt
package com.smlnskgmail.jaman.remotetemperaturecontrol.datasend.data

import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.entities.signaltype.SignalTarget
import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.entities.signaltype.SignalType
import com.smlnskgmail.jaman.remotetemperaturecontrol.monitor.support.MonitorTarget

class TestSignalTarget(private val monitorTarget: MonitorTarget) : SignalTarget {

    override fun onNewDataAvailable(signalType: SignalType, data: String) {
        when (signalType) {
            SignalType.Temperature -> monitorTarget.temperatureAvailable(data)
            SignalType.TemperatureMinimum -> monitorTarget.temperatureMinimumAvailable(data)
            SignalType.TemperatureMaximum -> monitorTarget.temperatureMaximumAvailable(data)
            SignalType.Humidity -> monitorTarget.humidityAvailable(data)
            SignalType.HumidityMinimum -> monitorTarget.humidityMinimumAvailable(data)
            SignalType.HumidityMaximum -> monitorTarget.humidityMaximumAvailable(data)
            SignalType.Reset -> monitorTarget.resetRequired()
        }
    }

}
